### PR TITLE
#51 - Configure api for plans (Backend)

### LINKED ISSUE #51: Configure api for plans (Backend)

### PR DESCRIPTION
**Pull request by NN171**

### CODE REVIEW

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java [hunk 1]
```diff
@@ -25,18 +24,23 @@ public ResponseEntity<String> parseProfiles(@RequestParam String url) {
...
-    @GetMapping("/")
-    public ResponseEntity<List<ProfileDto>> getAllProfiles() {
-        return ResponseEntity.ok(new ArrayList<>());
+    @GetMapping("/institute")
+    public ResponseEntity<List<ProfileDto>> getProfilesByInstitute(@RequestParam String institute) {
+        return ResponseEntity.ok(profileService.getProfilesByInstitute(institute));
+    }
+
+    @GetMapping("/direction")
+    public ResponseEntity<List<ProfileDto>> getProfilesByDirection(@RequestParam String code) {
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java]: Recommend to rename `getProfilesByDirection` to `getProfilesByDirectionCode` for increasing understanding

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java [hunk 2]
```diff
@@ -25,18 +24,23 @@ public ResponseEntity<String> parseProfiles(@RequestParam String url) {
...
+        return ResponseEntity.ok(profileService.getProfilesByDirection(code));
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java]: The same

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java [hunk 3]
```diff
@@ -25,18 +24,23 @@ public ResponseEntity<String> parseProfiles(@RequestParam String url) {
...
     }
 
-    @GetMapping("/pass")
-    public ResponseEntity<List<ProfileDto>> getProfilesByPassPoint(@PathVariable int passPoint) {
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java:34]: Profiles in one direction have different pass points. Why did you remove this method?
What's I mean: direction combines several profiles. Each profile have pass point and the direction get the minimum pass point from the profiles.
Correct me if I'm wrong

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/model/domain/AnnotationData.java [hunk 4]
```diff
@@ -1,3 +1,4 @@
 package com.example.BackendMIIT.model.domain;
 
-public record AnnotationData(String link, int year) {}
+public record AnnotationData(String link, int year) {
+}
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/model/domain/AnnotationData.java]: It doesn't look nice

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/EducationPlanService.java [hunk 5]
```diff
@@ -1,6 +1,10 @@
 package com.example.BackendMIIT.service;
 
+import com.example.BackendMIIT.model.dto.EducationPlanDto;
+
 public interface EducationPlanService {
 
     void parseAndSaveEducationPlans();
+
+    EducationPlanDto getPlansByProfile(String name);
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/EducationPlanService.java]: Recommend to rename `getPlansByProfile` to `getPlansByProfileName`

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/ProfileService.java [hunk 6]
```diff
@@ -10,4 +9,12 @@ public interface ProfileService {
...
 
     void saveProfile(List<String> properties);
+
+    List<ProfileDto> getProfilesByInstitute(String institute);
+
+    List<ProfileDto> getAllProfiles();
+
+    ProfileDto getProfileByName(String name);
+
+    List<ProfileDto> getProfilesByDirection(String code);
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/ProfileService.java]: Recommend to rename `getProfilesByDirection` to `getProfilesByDirectionCode`

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/DirectionServiceImpl.java [hunk 7]
```diff
@@ -17,9 +20,34 @@
...
-    public DirectionServiceImpl(DirectionRepository directionRepository) {
+    public DirectionServiceImpl(DirectionRepository directionRepository, DirectionMapper directionMapper) {
         this.directionRepository = directionRepository;
+        this.directionMapper = directionMapper;
+    }
+
+    @Override
+    public DirectionDto getDirectionByName(String name) {
+        Direction direction = directionRepository.findByName(name)
+                .orElseThrow(() -> new EntityNotFoundException("Direction doesn't exist"));
```
Comment (ImmortalIdiot) [inline backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/DirectionServiceImpl.java:33]: If we didn't catch `EntityNotFoundException` the application is crushed?

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java (hunk 3 above)
Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java:34]: Profiles don't have any pass points. Only directions contains them

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/DirectionServiceImpl.java (hunk 7 above)
Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/DirectionServiceImpl.java:33]: No, because parent of this exception is RuntimeException

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/ProfileService.java (hunk 6 above)
Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/ProfileService.java]: Renamed

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/EducationPlanService.java (hunk 5 above)
Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/EducationPlanService.java]: Renamed

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/model/domain/AnnotationData.java (hunk 4 above)
Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/model/domain/AnnotationData.java]: Changed

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java (hunk 2 above)
Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java]: Renamed

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java (hunk 1 above)
Comment (NN171) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java]: Renamed

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java (hunk 3 above)
Comment (ImmortalIdiot) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/controller/ProfileController.java:34]: Ok

File: backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/DirectionServiceImpl.java (hunk 7 above)
Comment (ImmortalIdiot) [reply backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/DirectionServiceImpl.java:33]: If RuntimeException didn't catch then the application is crushed

**Review by ImmortalIdiot — APPROVED** [review]
